### PR TITLE
Test for unicast validation

### PR DESCRIPTION
**Pull request by alonSadan**
A test for  the function checkCast(), which is used when creating
a new pool manager to verify that the first octet of the ranges
is 02,06,0A or 0E.

### CODE REVIEW

File: pkg/pool-manager/pool_test.go [hunk 1]
```diff
@@ -64,20 +64,48 @@ var _ = Describe("Pool", func() {
...
 			startPoolRangeEnv, err := net.ParseMAC("02:00:00:00:00:00")
 			Expect(err).ToNot(HaveOccurred())
-			endPoolRangeEnv, err := net.ParseMAC("FD:FF:FF:FF:FF:FF")
+			endPoolRangeEnv, err := net.ParseMAC("02:FF:FF:FF:FF:FF")
 			Expect(err).ToNot(HaveOccurred())
 			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
 			Expect(err).ToNot(HaveOccurred())
 		})
 
 		It("should fail to pool manager because of the invalid range", func() {
```
Comment (SchSeba) [inline pkg/pool-manager/pool_test.go]: should fail to **create** a pool manager object when the range is invalid

File: pkg/pool-manager/pool_test.go [hunk 2]
```diff
@@ -64,20 +64,48 @@ var _ = Describe("Pool", func() {
...
+			fakeClient := fake.NewSimpleClientset()
+			startPoolRangeEnv, err := net.ParseMAC("0A:00:00:00:00:00")
+			Expect(err).ToNot(HaveOccurred())
+			endPoolRangeEnv, err := net.ParseMAC("02:00:00:00:00:00")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should fail to pool manager because of the first octet of RangeStart is not 02, 06, 0A, 0E", func() {
```
Comment (SchSeba) [inline pkg/pool-manager/pool_test.go]: should fail to create a pool manager object when the first octet of RangeStart is not 02, 06, 0A, 0E

File: pkg/pool-manager/pool_test.go [hunk 3]
```diff
@@ -86,6 +114,17 @@ var _ = Describe("Pool", func() {
 			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
 			Expect(err).To(HaveOccurred())
 		})
+
+		It("should fail to pool manager because of the first octet of RangeEnd is not 02, 06, 0A, 0E", func() {
```
Comment (SchSeba) [inline pkg/pool-manager/pool_test.go]: should fail to create a pool manager object when the first octet of RangeEnd is not 02, 06, 0A, 0E

File: pkg/pool-manager/pool_test.go [hunk 4]
```diff
@@ -86,6 +114,17 @@ var _ = Describe("Pool", func() {
...
+			fakeClient := fake.NewSimpleClientset()
+			startPoolRangeEnv, err := net.ParseMAC("02:00:00:00:00:00")
+			Expect(err).ToNot(HaveOccurred())
+			endPoolRangeEnv, err := net.ParseMAC("05:00:00:00:00:00")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
+			Expect(err).To(HaveOccurred())
```
Comment (SchSeba) [inline pkg/pool-manager/pool_test.go]: please check that the error string is the expected one

```
Invalid mac address. The first octet should be: 02, 06, 0A or 0E
```

File: pkg/pool-manager/pool_test.go [hunk 5]
```diff
@@ -64,20 +64,48 @@ var _ = Describe("Pool", func() {
...
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		},
+			table.Entry("Valid address: 02:00:00:00:00:00", "02:00:00:00:00:00", false),
+			table.Entry("Valid address: 06:00:00:00:00:00", "06:00:00:00:00:00", false),
+			table.Entry("Valid address: 0A:00:00:00:00:00", "0A:00:00:00:00:00", false),
+			table.Entry("Valid address: 0E:00:00:00:00:00", "0E:00:00:00:00:00", false),
+			table.Entry("Invalid address: 01:FF:00:00:00:00, the first octet is not 02, 06, 0A or 0E", "01:00:00:00:00:00", true),
```
Comment (HlinaCZ) [inline pkg/pool-manager/pool_test.go]: You have mismatched address in title and function parameter.

File: cluster/k8s-1.13.3/.kubeconfig [hunk 6]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
```
Comment (phoracek) [inline cluster/k8s-1.13.3/.kubeconfig]: should not  be commited. Isn't .kubeconfig a part of gitignore?

File: config/default/manager/manager.yaml [hunk 7]
```diff
@@ -17,7 +17,7 @@ metadata:
     controller-tools.k8s.io: "1.0"
 data:
   RANGE_START: 02:00:00:00:00:00
-  RANGE_END: FD:FF:FF:FF:FF:FF
+  RANGE_END: 02:FF:FF:FF:FF:FF
```
Comment (phoracek) [inline config/default/manager/manager.yaml]: is this related to the test?

File: pkg/pool-manager/pool.go [hunk 8]
```diff
@@ -146,7 +154,16 @@ func checkRange(startMac, endMac net.HardwareAddr) error {
 		}
 	}
 
-	return fmt.Errorf("Invalid range start: %s end: %s", startMac.String(), endMac.String())
+	return fmt.Errorf("Invalid range. rangeStart: %s rangeEnd: %s", startMac.String(), endMac.String())
+}
+
+func checkCast(mac net.HardwareAddr) error {
+	s := mac[0]
+	if s == 0x2 || s == 0x6 || s == 0xA || s == 0xE {
```
Comment (phoracek) [inline pkg/pool-manager/pool.go]: Is it about octet though? AFAIK unicast is signalized by a single bit. Can't you just match that bit? As wiki says "When the least significant bit of an address's first octet is 0 (zero), the frame is meant to reach only one receiving NIC.". Could you & it with 1, if the result is 1, we are multicast. It will also make tests shorter.

File: pkg/pool-manager/pool_test.go [hunk 9]
```diff
@@ -64,27 +64,70 @@ var _ = Describe("Pool", func() {
 			table.Entry("Start: 02:00:00:00:00:00  End: 02:10:00:00:00:00", "02:00:00:00:00:00", "02:10:00:00:00:00", false),
 			table.Entry("Start: 02:FF:00:00:00:00  End: 02:00:00:00:00:00", "02:FF:00:00:00:00", "00:00:00:00:00:00", true),
 		)
+
+		table.DescribeTable("should check that the the first octet is 02, 06, 0A or 0E", func(MacAddr string, needToFail bool) {
```
Comment (phoracek) [inline pkg/pool-manager/pool_test.go]: shouldFail

File: config/default/manager/manager.yaml (hunk 7 above)
Comment (alonSadan) [reply config/default/manager/manager.yaml]: @phoracek  Thanks for the notes. It just that in FD:FF:FF.... the multicast bit is on

File: cluster/k8s-1.13.3/.kubeconfig (hunk 6 above)
Comment (SchSeba) [reply cluster/k8s-1.13.3/.kubeconfig]: it is there is a problem with the commits on this PR I think he should rebase

**Comment by alonSadan** [comment]
please ignore this PR till I fix the commits...
